api/v1alpha2/awsmachine_conversion.go

```diff
@@ -19,19 +19,46 @@ package v1alpha2
 import (
 	apiconversion "k8s.io/apimachinery/pkg/conversion"
 	infrav1alpha3 "sigs.k8s.io/cluster-api-provider-aws/api/v1alpha3"
+	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
 	"sigs.k8s.io/controller-runtime/pkg/conversion"
 )
 
 // ConvertTo converts this AWSMachine to the Hub version (v1alpha3).
 func (src *AWSMachine) ConvertTo(dstRaw conversion.Hub) error { // nolint
 	dst := dstRaw.(*infrav1alpha3.AWSMachine)
-	return Convert_v1alpha2_AWSMachine_To_v1alpha3_AWSMachine(src, dst, nil)
+	if err := Convert_v1alpha2_AWSMachine_To_v1alpha3_AWSMachine(src, dst, nil); err != nil {
+		return err
+	}
+
+	// Manually restore data.
+	restored := &infrav1alpha3.AWSMachine{}
+	if ok, err := utilconversion.UnmarshalData(src, restored); err != nil || !ok {
+		return err
+	}
+	restoreAWSMachineSpec(&restored.Spec, &dst.Spec)
+
+	return nil
+}
+
+func restoreAWSMachineSpec(restored *infrav1alpha3.AWSMachineSpec, dst *infrav1alpha3.AWSMachineSpec) {
+	if restored.RootVolumeOptions != nil {
+		dst.RootVolumeOptions = restored.RootVolumeOptions
+	}
```

Comment:
We can probably use `dst.RootVolumeOptions.DeepCopyInto(...)`

Reply:
Also note that this would overwrite the Manual conversion for `RootDeviceSize` that we have in `Convert_v1alpha2_AWSMachineSpec_To_v1alpha3_AWSMachineSpec`